src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1315,15 +1382,18 @@
         {:keys [limit offset order-by]} (create-paging-map clauses)]
     (if (extract-expression? expr)
       (let [[extract columns remaining-expr] expr
-            column-list (utils/vector-maybe columns)]
+            column-list (utils/vector-maybe columns)
+            projections (:projections query-rec)]
         (-> query-rec
-            (assoc :projected-fields column-list :where (user-node->plan-node query-rec remaining-expr))
+            (assoc :projected-fields (mapv (fn [name] [name (projections name)])
+                                           column-list)
```

Comment:
Looks like you have a similar function above, might be worth pulling giving it a name and docstring :)